File: 2379. Minimum Recolors to Get K Consecutive Black Blocks.cpp
class Solution {
    public:
        int minimumRecolors(string blocks, int k) 
        {
            int cnt = 0, ans = INT_MAX;
            int i = 0, j = 0, n = blocks.size();
            while(j < n)
            {
                if(blocks[j++] == 'W') cnt++;
                while(j - i > k)
                    if(blocks[i++] == 'W') cnt--;
                if(j - i == k) ans = min(ans, cnt);
            }
            return ans;
        }
    };

File: 2379. Minimum Recolors to Get K Consecutive Black Blocks.js
var minimumRecolors = function(blocks, k) 
{
    let n = blocks.length;
    let i = 0, j = 0, ans = k, count = 0;    
    while(j < n)
    {
        count += blocks[j++] == 'W';
        if(j - i == k) 
        {
            ans = Math.min(ans, count);
            count -= blocks[i++] == 'W';
        }
    }
    return ans;
};